include/tunnler/room_member.h

```diff
@@ -23,6 +24,14 @@
 // It also has to be used if you host a game yourself (You'd create both, a Room and a RoomMembership for yourself)
 class RoomMember final {
 public:
+    using Callback = std::shared_ptr<std::function<void()> >;
```

Comment:
As I see it, you prefer copying around pointers instead of the function classes, thus shared_ptr is the right way.